src/todos.rs

```diff
@@ -22,32 +27,57 @@ impl Todos {
     }
 }
 
+#[async_trait]
 impl TodoStorage for Todos {
-    fn add_todo(&mut self, todo: Todo) {
+    async fn read_storage(&mut self) {
+        let file = read_to_string("session.txt").await;
+        match file {
+            Ok(file) => {
+                let mut text_todos: Vec<&str> = file.split("-").collect();
+                text_todos.pop();
+                let mut count = 0;
+                while count < text_todos.len() {
+                    let todo = Todo::new(text_todos[count].to_string(), text_todos[count + 1].contains("true"));
+                    self.todos.push(todo);
+                    count +=2
+                }
```

Comment:
A lógica que vc fez com o while já é implementada por [`[T]::chunks_exact`](https://doc.rust-lang.org/std/primitive.slice.html#method.chunks_exact).

```suggestion
                for chunk in text_todos.chunks_exact(2) {
                    let todo = Todo::new(chunk[0].to_string(), chunk[1].contains("true"));
                    self.todos.push(todo);
                }
```

Reply:
Pronto mega!!